test/Tests/src/System/File_Spec.enso

```diff
@@ -126,6 +126,10 @@ spec =
             contents_2 = File.new file . read_bytes
             contents_2.take (First 6) . should_equal [67, 117, 112, 99, 97, 107]
 
+        Test.specify "no parent of dot implies nothing" <|
+            no_parent = (File.new '.').parent
+            no_parent.is_nothing . should_be_true
```

Comment:
I'm glad we've got a test documenting the current behaviour and that it does not break.

Possibly for a separate PR but I'm wondering if that should be the behaviour though? My intuition tells me that `'.'.parent == './..'` since this would make the following commute: `'.'.absolute.parent == '.'.parent.absolute`. And it just seems more intuitive and useful to me.

Still, I'm not sure how hard would it be to amend the current implementation to achieve that.

Reply:
I've only put the issue into _design stage_ (not _implementation_) as it needs some API love anyway. Thanks for your comments Radek. 

> I'm glad we've got a test documenting the current behaviour and that it does not break.

Right, I believe a non-`File` return value is needed. At least for `/` which clearly doesn't have a `.parent`, but I can also return an `Error` if that's want you'd rather see.

> Possibly for a separate PR ...

I'd say there are some problems with the current behavior. For example, if `(File.new '.')` is
```
/enso/enso/.
```
then its `.parent` is:
```
/enso/enso
```
that's wrong, right? `.` means the current directory, not a `.` file in the current directory?

Anyway I happily leave the library maintainers to deal with other consequences and I just change `File.parent : File | Nothing` or `File ! Error`. Just let me know.

Reply:
Agree with your example @JaroslavTulach - yes parent of `/enso/enso/,` should be `/enso`.

Reply:
> Agree with...

OK, here is the issue #5703